web/app/js/components/ResourceDetail.jsx

```diff
@@ -298,6 +304,15 @@ export class ResourceDetailBase extends React.Component {
 
     let showNoTrafficMsg = resourceIsMeshed && (Date.now() - lastMetricReceivedTime > showNoTrafficMsgDelayMs);
 
+    if (isTrafficSplit) {
+      return (
+        <TrafficSplitDetail
+          resourceType={resourceType}
+          resourceName={resourceName}
+          resourceMetrics={resourceMetrics}
+          resourceRsp={resourceRsp} />
```

Comment:
tioli: this `TrafficSplitDetail` block and the `<div>` below feel a bit incongruous, as the former leverages a new component while the latter is in-lined. also, a lot of the code immediately above `TrafficSplitDetail` is not relevant if `isTrafficSplit` is true. consider breaking out the `div` below into a new component. this would have the added bonus of moving a lot of the code that is only relevant to the `!isTrafficSplit` case into that new component.

Reply:
I see your point. My goal was to change `ResourceDetail` as little as possible, so I built the `TrafficSplitDetail` component as an extension of `ResourceDetail`. But this does mean that there is a lot of code in `ResourceDetail` that is not relevant to trafficsplits. Let me experiment with reorganizing to see if it’s more clear.